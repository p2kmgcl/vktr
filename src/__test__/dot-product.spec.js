import dotProduct from "../dot-product";
import Vector from "../vector";

describe("dotProduct()", () => {
  it("returns the dot product of two vectors", () => {
    const a = new Vector(1, 1);
    const b = new Vector(-1, -1);

    expect(dotProduct(a, b)).toMatchSnapshot();
  });

  it("require them to have same number of dimensions", () => {
    expect(() => {
      dotProduct(new Vector(1, 2, 3), new Vector(3, 4));
    }).toThrow(RangeError);
  });

  it("is commutative", () => {
    const a = new Vector(1, 1);
    const b = new Vector(-1, -1);

    expect(dotProduct(a, b)).toEqual(dotProduct(b, a));
  });

  it("works for n-dimensional vectors", () => {
    const a = new Vector(1, 1, 3, 7, 13);
    const b = new Vector(-1, -1, 0, 25, 1);

    expect(dotProduct(a, b)).toMatchSnapshot();
  });
});
