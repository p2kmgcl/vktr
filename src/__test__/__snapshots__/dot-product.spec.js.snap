// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotProduct() returns the dot product of two vectors 1`] = `-2`;

exports[`dotProduct() works for n-dimensional vectors 1`] = `186`;
